drivers/cc1200/cc1200-netdev.c

```diff
@@ -0,0 +1,289 @@
+        //return cc1200_pkt->length - IEEE802154_FCS_LEN;
+        return cc1200_pkt->length;
+    }
+
+
+    if (cc1200_pkt->length > len) {
+        return -ENOSPC;
+    }
+
+    memcpy(buf, (void *)cc1200_pkt + 1, cc1200_pkt->length);
+    if (info != NULL) {
+        netdev_cc1200_rx_info_t *cc1200_info = info;
+
+        cc1200_info->rssi = cc1200->pkt_buf.rssi;
+        cc1200_info->lqi = cc1200->pkt_buf.lqi;
+    }
+    DEBUG("%s:%u pkt->length: %u\n", __func__, __LINE__, cc1200_pkt->length);
+#ifdef MODULE_NETSTATS_L2
+    dev->stats.rx_count++;
+    dev->stats.rx_bytes += cc1200_pkt->length;
+#endif
+
+    for (uint8_t i = 0; i < cc1200_pkt->length + 1; i++)
+        DEBUG("0x%x ", *(((char * )cc1200_pkt) + i));
+    DEBUG("\n");
+
+    return cc1200_pkt->length;
+}
+
+
+static int _get(netdev_t *dev, netopt_t opt, void *value, size_t value_len)
+{
+
+    int res = -ENOTSUP;
+
+    switch (opt) {
+
+        case NETOPT_MAX_PACKET_SIZE:
+            assert(value_len >= sizeof(int16_t));
+            *((uint16_t *)value) = CC1200_MAX_DATA_LENGTH - _MAX_MHR_OVERHEAD;
+            return sizeof(uint16_t);
+
+        default:
+            break;
+    }
+
+    if (((res = netdev_ieee802154_get((netdev_ieee802154_t *)dev, opt, value,
+                                      value_len)) >= 0) || (res != -ENOTSUP)) {
+        return res;
+    }
+    return res;
+}
+
+static int _set(netdev_t *dev, netopt_t opt, const void *value, size_t value_len)
+{
+    cc1200_t *cc1200 = &((netdev_cc1200_t *) dev)->cc1200;
+    int res = -ENOTSUP;
+
+    switch (opt) {
+        case NETOPT_CHANNEL:
+        {
+            if (value_len != sizeof(uint16_t)) {
+                return -EINVAL;
+            }
+            uint8_t channel = (((const uint16_t *)value)[0]) & UINT8_MAX;
+            if (channel > CC1200_MAX_CHANNR) {
+                return -EINVAL;
+            }
+            if (cc1200_set_channel(cc1200, channel) == -1) {
+                return -EINVAL;
+            }
+            break;
+        }
+        case NETOPT_ADDRESS:
+            if (value_len < 1) {
+                return -EINVAL;
+            }
+            if (!cc1200_set_address(cc1200, *(uint8_t *)value)) {
+                return -EINVAL;
+            }
+            if (!cc1200_set_address_short(cc1200, *(uint16_t *)value)) {
+                return -EINVAL;
+            }
+            break;
+        case NETOPT_NID:
+            if (value_len > sizeof(uint16_t)) {
+                res = -EOVERFLOW;
+            }
+            if (value_len < 1) {
+                return -EINVAL;
+            }
+            cc1200->pan_id = *(uint16_t *) value;
+            break;
+        case NETOPT_ADDRESS_LONG:
+            if (value_len > sizeof(uint64_t)) {
+                res = -EOVERFLOW;
+            }
+            else {
+                res = cc1200_set_address_long(cc1200, *((uint64_t *)value));
+            }
+            break;
+#ifdef MODULE_GNRC_NETIF
+        case NETOPT_PROTO:
```

Comment:
I think this is already handled by `netdev_ieee802154` [here](https://github.com/RIOT-OS/RIOT/blob/master/drivers/netdev_ieee802154/netdev_ieee802154.c#L215), isn't it?